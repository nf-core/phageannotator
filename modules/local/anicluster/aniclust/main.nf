process ANICLUSTER_ANICLUST {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' :
        'biocontainers/mulled-v2-80c23cbcd32e2891421c54d1899665046feb07ef:77a31e289d22068839533bf21f8c4248ad274b60-0' }"

    input:
    tuple val(meta), path(fasta), path(ani)

    output:
    tuple val(meta), path("*_clusters.tsv") , emit: clusters
    path "versions.yml"                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    aniclust.py \\
        --fna $fasta \\
        --ani $ani \\
        --out ${prefix}_clusters.tsv \\
        $args

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_clusters.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
    END_VERSIONS
    """
}
