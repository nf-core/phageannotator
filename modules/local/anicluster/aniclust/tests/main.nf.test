nextflow_process {

    name "Test process: ANICLUSTER_ANICLUST"
    script "../main.nf"
    process "ANICLUSTER_ANICLUST"


    test("fasta.gz & ani.tsv") {

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists: true ),
                    file(params.pipelines_testdata_base_path + 'modules/local/anicluster/aniclust/ani.tsv', checkIfExists: true )
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("fasta.gz & ani.tsv - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ],
                    file(params.modules_testdata_base_path + 'genomics/prokaryotes/bacteroides_fragilis/illumina/fasta/test1.contigs.fa.gz', checkIfExists: true ),
                    file(params.pipelines_testdata_base_path + 'modules/local/anicluster/aniclust/ani.tsv', checkIfExists: true )
                ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
