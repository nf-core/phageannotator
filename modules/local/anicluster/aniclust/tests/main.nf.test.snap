{
    "fasta.gz & ani.tsv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_clusters.tsv:md5,adcc2d4d4b096d6c9640180e94dd6c5f"
                    ]
                ],
                "1": [
                    "versions.yml:md5,774a3132a87e17391bf8b7ecf92c58d3"
                ],
                "clusters": [
                    [
                        {
                            "id": "test"
                        },
                        "test_clusters.tsv:md5,adcc2d4d4b096d6c9640180e94dd6c5f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,774a3132a87e17391bf8b7ecf92c58d3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-19T17:10:08.270406311"
    },
    "fasta.gz & ani.tsv - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_clusters.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,774a3132a87e17391bf8b7ecf92c58d3"
                ],
                "clusters": [
                    [
                        {
                            "id": "test"
                        },
                        "test_clusters.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,774a3132a87e17391bf8b7ecf92c58d3"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-19T17:10:16.994335689"
    }
}
